lib/tf_lock/tf_lock_acquire.py

```diff
@@ -62,6 +90,37 @@ def tf_lock_acquire(root_module: Path) -> ExitCode:
         # start over
 
 
+def is_pr_closed(org_name: str, repo_name: str, pr_number: str) -> bool:
+    import json
+    import os
+    from urllib.error import HTTPError
+    from urllib.request import Request
+    from urllib.request import urlopen
+
+    access_token = os.getenv("GH_TOKEN")
+    api_url = f"https://api.github.com/repos/{org_name}/{repo_name}/pulls/{pr_number}"
+    headers = {
+        "Authorization": f"token {access_token}",
+        "Accept": "application/vnd.github.v3+json",
+    }
+    req = Request(api_url, headers=headers)
+
+    try:
+        with urlopen(req) as response:
+            pr_data = json.loads(response.read().decode())
+            if pr_data["state"] == "closed":
+                return True
+    except HTTPError as e:
+        if e.code == 404:
+            raise ValueError(
+                "PR not found or you do not have access to this repository."
+            )
+        else:
+            raise Exception(f"Failed to fetch PR details: {e.code}")
+
+    return False
+
+
```

Comment:
Is this the same as https://github.com/getsentry/tacos-gha/blob/main/spec/lib/gh/pr.py#L107-L111 but implemented differently not using the `gh` CLI.

I know the other is part of `spec` but it would be good to consolidate/avoid building this functionality multiple times in different ways